app/services/hesa/backfill_degrees.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module Hesa
+  class BackfillDegrees < Backfill
+    def call
+      return unless trainee_trns.any?
+
+      Trainee.where(trn: trainee_trns).find_each do |trainee|
+        Degree.without_auditing do
+          ::Degrees::CreateFromHesa.call(
+            trainee: trainee,
+            hesa_degrees: trainees_with_degrees[trainee.trn],
+          )
+        end
+      end
+    end
+
+  private
+
+    def trainees_with_degrees
+      return @trainees_with_degrees if @trainees_with_degrees
```

Comment:
Something like this could make the intent a bit clearer:
```suggestion
    def trainees_with_degrees
      @trainees_with_degrees ||= build_trainees_with_degrees
    end
    
    def build_trainees_with_degrees
```